### PR TITLE
Add configure model page

### PR DESCRIPTION
**Pull request by ohltyler**
*Issue #, if available:*

*Description of changes:*

**This PR is 2 of 9 related to the new single flow changes. We will merge all changes into the single-flow-development branch 
first, and finally merge into main.**

**Please ignore test-related code for now, this is currently being refactored and updated, will be addressed in a later PR**

This PR adds the configure model page (step 2 in the creation flow). Specifically, the changes include:
- Adds the new `ConfigureModel` page, which is mostly a refactored & renamed version of the current `EditFeatures` page. The motivation for creating this new page was to make it clear that this page is about all model configuration settings, rather than just the created features, as well as make the layout and logic consistent with the other create flow pages for easier future maintainability and readability. Also note that the functionality of the components themselves (feature list, advanced settings, preview) have not changed, just moved to this new directory.
- Adds new props to `ConfigureModel` container compared to `EditFeatures` which makes it consistent with the rest of the new create flow pages (`setStep()`, `setInitialValues`, etc.)
- Update and move the helper functions to convert detector fields to Formik and vice versa
- Changes validation to be consistent with the rest of the new create flow pages

Screenshot:
screencapture-localhost-5601-app-opendistro-anomaly-detection-kibana-2021-04-05-11_08_32

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/ConfigureModel/containers/ConfigureModel.tsx [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+            {props.isEdit ? null : (
+              <EuiFlexItem grow={false}>
+                <EuiButton
+                  iconSide="left"
+                  iconType="arrowLeft"
+                  fill={false}
+                  data-test-subj="configureModelPreviousButton"
+                  //isLoading={formikProps.isSubmitting}
```
Comment (yizheliu-amazon) [inline public/pages/ConfigureModel/containers/ConfigureModel.tsx]: commented line.

File: public/pages/ConfigureModel/utils/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+  const initialValues = cloneDeep(INITIAL_MODEL_CONFIGURATION_VALUES);
+  if (isEmpty(detector)) {
+    return initialValues;
+  }
+  return {
+    ...initialValues,
+    featureList: featuresToFormik(detector),
+    categoryFieldEnabled: !isEmpty(get(detector, 'categoryField', []))
+      ? true
+      : false,
```
Comment (yizheliu-amazon) [inline public/pages/ConfigureModel/utils/helpers.ts]: we can just use `!isEmpty(get(detector, 'categoryField', []))`?

**Review by yizheliu-amazon — APPROVED** [review]

File: public/pages/ConfigureModel/containers/ConfigureModel.tsx (hunk 1 above)
Comment (ohltyler) [reply public/pages/ConfigureModel/containers/ConfigureModel.tsx]: will remove

File: public/pages/ConfigureModel/utils/helpers.ts (hunk 2 above)
Comment (ohltyler) [reply public/pages/ConfigureModel/utils/helpers.ts]: good catch, will remove

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**